pkg/config/nodetreemodel/config_test.go

```diff
@@ -18,7 +22,7 @@ import (
 )
 
 // Test that a setting with a map value is seen as a leaf by the nodetreemodel config
-func TestBuildDefaultMakesTooManyNodes(t *testing.T) {
+func TestLeafNodeCanHaveComplexMapValue(t *testing.T) {
```

Comment:
This test name was chosen in #30361 because it was describing an incorrect behavior: that settings that have map values used to be turned into trees. Instead they should be a leaf nodes with complex map values. This was fixed by #30746, which removed the `t.Skip` from this test, but mistakingly did not rename it.